src/NHibernate/Criterion/ProjectionsExtensions.cs

```diff
@@ -70,7 +70,7 @@ internal static IProjection ProcessDate(System.Linq.Expressions.Expression expre
 		/// </summary>
 		public static double Sqrt(this double numericProperty)
 		{
-			throw new Exception("Not to be used directly - use inside QueryOver expression");
+			throw new InvalidOperationException("Not to be used directly - use inside QueryOver expression");
```

Comment:
I think it's better to wrap this exception creation in some method to minimize copy-paste for future extensions/refactorings. So it should look something like:
`throw DirectUsageException();`

Reply:
Does it have to throw at all? Real methods are replaced in queries all the time, so why not here, i.e call Math.Sqrt and be done with it?

Reply:
@gliljas could do, but in another RP.

Reply:
> Does it have to throw at all? Real methods are replaced in queries all the time, so why not here, i.e call Math.Sqrt and be done with it?

Some cases could be trickier, like the `Projection.Concat` where `string.Concat` `null` handling will not be similar to SQL default `null` handling. So this may cause more troubles than simply throwing.